observatory-platform/observatory/platform/utils/gc_utils.py

```diff
@@ -1232,3 +1234,136 @@ def delete_bucket_dir(*, bucket_name: str, prefix: str):
 
     for blob in blobs:
         blob.delete()
+
+
+def list_all_buckets():
+    """List of all Google Cloud buckets.
+
+    :return: list of buckets that the service account as access to under the project.
+    """
+
+    # Client to read Google Cloud Storage
+    storage_client = storage.Client()
+    buckets = storage_client.list_buckets()
+
+    # Make list of all names of the storage buckets
+    bucket_list = []
+    for bucket in buckets:
+        bucket_list.append(bucket.name)
+
+    return bucket_list
+
+
+def list_all_datasets():
+    """List of all BigQuery datasets.
+
+    :return: list of datasets that the service account as access to under the project.
+    """
+
+    client = bigquery.Client()
+    datasets = list(client.list_datasets())
+
+    dataset_list = []
+    for dataset in datasets:
+        dataset_list.append(dataset.dataset_id)
+
+    return dataset_list
```

Comment:
I would also try to consolidate `list_all_datasets` and `get_age_of_dataset_in_days`, i.e. make `list_all_datasets` return a list of `Dataset` objects, and then put the time difference calculation in `delete_old_datasets_with_prefix` (which calls calls `dataset.created`).